### DISCUSSION TITLE
Cleaning up cache

### DISCUSSION THREAD

**Discussion by eidolonFIRE** [JaffaKetchup/flutter_map_tile_caching#119] [Q&A] [answered]
In v6, I was able to clean up old files from a `store` using the following code.
( [old discussion ](https://github.com/JaffaKetchup/flutter_map_tile_caching/discussions/62))

```
void purgeMapTileCache() async {
  final threshhold = DateTime.now().subtract(const Duration(days: 14));
  for (final tileSrc in mapTileThumbnails.keys) {
    final tileName = tileSrc.toString().split(".").last;
    final StoreDirectory store = FMTC.instance(tileName);

    int countDelete = 0;
    int countRemain = 0;
    for (final tile in store.access.tiles.listSync()) {
      if (tile.statSync().changed.isBefore(threshhold)) {
        // debugPrint("Deleting Tile: ${tile.path}");
        tile.deleteSync();
        countDelete++;
      } else {
        countRemain++;
      }
    }
    debugPrint("Scanned $tileName and deleted $countDelete / ${countRemain + countDelete} tiles.");
    store.stats.invalidateCachedStatistics();
  }
}
```

It looks like in v7 a couple things are gone and I haven't been able to track down a 1:1 transition. I'm not sure if my approach in v6 even makes sense with the new database.

How should I be purging old tiles in v7?

**Comment by JaffaKetchup** [comment, marked as answer]
Hi @eidolonFIRE,
I'll consider this a feature request and add the functionality for the next version. L
You can do it yourself if you're comfortable with Isar.
